File: Check_Path.c
#include <stdio.h>
#include <stdbool.h>

void addEdge(int V, int adj[V][V], int u, int v) {
    adj[u][v] = 1;
    adj[v][u] = 1; // For an undirected graph
}

bool hasPath(int V, int adj[V][V], int src, int dest) {
    bool visited[V];
    for (int i = 0; i < V; i++) {
        visited[i] = false;
    }

    int stack[V];
    int top = -1;

    visited[src] = true;
    stack[++top] = src;

    while (top >= 0) {
        int current = stack[top--];

        for (int neighbor = 0; neighbor < V; neighbor++) {
            if (adj[current][neighbor] == 1) {
                if (neighbor == dest) {
                    return true; // Path found
                }
                if (!visited[neighbor]) {
                    visited[neighbor] = true;
                    stack[++top] = neighbor;
                }
            }
        }
    }

    return false; // No path found
}

File: Check_Path.py
class Graph:
    def __init__(self, V):
        self.V = V
        self.adjList = [[] for _ in range(V)]

    def addEdge(self, u, v):
        self.adjList[u].append(v)
        self.adjList[v].append(u)  # For an undirected graph

    def hasPath(self, src, dest):
        visited = [False] * self.V
        queue = []

        visited[src] = True
        queue.append(src)

        while queue:
            current = queue.pop(0)

            for neighbor in self.adjList[current]:
                if neighbor == dest:
                    return True  # Path found
                if not visited[neighbor]:
                    visited[neighbor] = True
                    queue.append(neighbor)

        return False  # No path found
